presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -543,11 +552,30 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
         Map<List<String>, ComputedStatistics> computedStatisticsMap = createComputedStatisticsToPartitionMap(computedStatistics, partitionColumnNames, columnTypes);
 
         // commit analyze to unpartitioned table
-        PartitionStatistics tableStatistics = createPartitionStatistics(session, columnTypes, computedStatisticsMap.get(ImmutableList.<String>of()), timeZone);
+        ConnectorTableMetadata metadata = getTableMetadata(session, tableHandle);
+        Set<ColumnStatisticMetadata> hiveSupportedStatistics = getHiveSupportedColumnStatistics(session, metadata);
+        PartitionStatistics tableStatistics = createPartitionStatistics(
+                session,
+                columnTypes,
+                computedStatisticsMap.get(ImmutableList.<String>of()),
+                hiveSupportedStatistics,
+                timeZone);
         metastore.updateTableStatistics(metastoreContext,
                 table.getDatabaseName(),
                 table.getTableName(),
                 oldStats -> updatePartitionStatistics(oldStats, tableStatistics));
```

Comment:
I wonder after we use puffin file to record NDV and total size in bytes for columns, could we just omit recording and reading the stats from HMS and directly adopt the values from Iceberg native? Is there any scenario where HIVE's statistical data is better?

Reply:
There could be external systems which may record statistics through the HMS outside of Presto/Iceberg, so I wanted to at least provide the option to use the HMS should that be the case. I agree that ideally everything would write directly to Iceberg table format and we could always use those statistics, but I don't think we can rely on that always being true.

Reply:
Sure, got it!